frontend/src/hooks/useCustomEvent.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from 'react';
+
+/**
+ * A hook that provides methods to emit and listen to custom DOM events.
+ * @param eventName The name of the event to listen for and dispatch.
+ */
+export const useCustomEvent = (
```

Comment:
this hook uses document as event listener that can propagate events across the UI

Reply:
I decided to extract this hook separately and compose in the useLastViewedProject